src/ucp/core/ucp_ep.c

```diff
@@ -1201,6 +1218,13 @@ static void ucp_ep_discard_lanes(ucp_ep_h ep, ucs_status_t discard_status)
         }
 
         ucs_debug("ep %p: discard uct_ep[%d]=%p", ep, lane, uct_ep);
+
+        if (!ucp_wireup_ep_test(uct_ep) ||
+            ((ucp_wireup_ep(uct_ep)->super.uct_ep != NULL) &&
+            ucp_wireup_ep(uct_ep)->super.is_owner)) {
+            discard_arg->unprogress_mask |= UCS_BIT(lane);
+        }
```

Comment:
how does it work in case if other peers are still connected?

Reply:
unprogress really happens when there is no peer left. we have wiface->activate_count to track that.

Reply:
ok but need to apply this logic only for client-server EPs since we may don't know about peers connected by worker address to iface with `UCT_IFACE_FLAG_CONNECT_TO_IFACE` capability

Reply:
done